### PR TITLE
pkg/nimble: have RIOT always initialize nimble timers

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR implements suggestions in https://github.com/RIOT-OS/RIOT/pull/16348#discussion_r665160617, the upstream PR has been adapted accordingly https://github.com/apache/mynewt-nimble/pull/996. The patch can be removed once that Pr is merged and upstream commit updated.

### Testing procedure

It can be tested with a simple compile test verifying that its not called when using `uwb-core` and that it is when not (e.g.: adding a #error)

- note called ` CFLAGS=-fms-extensions USEPKG="uwb-dw1000 uwb-core" make -C examples/nimble_gatt/ all -j3`

- called ` make -C examples/nimble_gatt/ all -j3`

```
/home/francisco/workspace/RIOT/pkg/nimble/contrib/nimble_riot.c: In function 'nimble_riot_init':
/home/francisco/workspace/RIOT/pkg/nimble/contrib/nimble_riot.c:109:6: error: #error
  109 |     #11852 
```

### CODE REVIEW

File: pkg/nimble/contrib/nimble_riot.c [hunk 1]
```diff
@@ -105,6 +105,11 @@ void nimble_riot_init(void)
     int res;
     (void)res;
 
+#if !IS_USED(MODULE_MYNEWT_CORE) && IS_ACTIVE(NIMBLE_CFG_CONTROLLER)
+    hal_timer_init(5, NULL);
+    os_cputime_init(32768);
```
Comment (benpicco) [inline pkg/nimble/contrib/nimble_riot.c:110]: What are those magic numbers?

Comment (fjmolinas) [reply pkg/nimble/contrib/nimble_riot.c:110]: The 5th timer is NRF_RTC0 in nimble/mynewt. And the frequency is the maximum possible frequency, but there are actually defines for this I'll address.

Comment (fjmolinas) [reply pkg/nimble/contrib/nimble_riot.c]: FIxed the magic values and added a comment on mynewt-core and mynewt-nimble for this so its a bit less obscure

**Review by benpicco — APPROVED** [review]

**Comment by fjmolinas** [comment]
Go!

**Pull request merged**